google/cloud/storage/lifecycle_rule.cc

```diff
@@ -62,6 +63,17 @@ LifecycleRuleAction LifecycleRule::SetStorageClass(std::string storage_class) {
   return LifecycleRuleAction{"SetStorageClass", std::move(storage_class)};
 }
 
+LifecycleRuleCondition LifecycleRule::CreatedBefore(std::string const& date) {
```

Comment:
Q: Do you think it's necessary to have this `CreatedBefore` overload that takes a string? I know we have it today, and users may be passing full time strings, which would fail to parse as a date here and it would throw/crash. This overload is avoiding a build break, but instead there might be a runtime exception/crash.

Since this PR is already a breaking change, I'm wondering if maybe we want to get rid of this overload taking a string and require the caller to pass us a `CivilDay`, which will push any string-to-date error handling onto the caller.

Just a thought. WDYT?

Reply:
I was about to push the commit to remove these.

Reply:
I was waiting for this removal, but let me know if you'd like approval before that, or want to do that in a separate PR.

Reply:
Sorry for the delay, I had to rebase and fix the API baseline.